pkg/clients/elasticsearch.go

```diff
@@ -0,0 +1,199 @@
+package clients
+
+import (
+	"bytes"
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+	"strings"
+
+	"github.com/bitly/go-simplejson"
+	log "github.com/sirupsen/logrus"
+)
+
+const (
+	contentTypeJSON = "application/json"
+)
+
+type MGetRequest struct {
+	Docs []MGetItem `json:"docs"`
+}
+
+type MGetItem struct {
+	Type string `json:"_type,omitempty"`
+	Id   string `json:"_id,omitempty"`
+}
+
+type MGetResponse struct {
+	Docs []MGetResponseItem `json:"docs"`
+}
+type MGetResponseItem struct {
+	Index   string                 `json:"_index,omitempty"`
+	Version int                    `json:"_version,omitempty"`
+	Found   bool                   `json:"found,omitempty"`
+	Source  map[string]interface{} `json:"_source,omitempty"`
+	MGetItem
+}
+
+//ElasticsearchClient is an admin client to query a local instance of Elasticsearch
+type ElasticsearchClient interface {
+	Get(path string) (string, error)
+	MGet(index string, items MGetRequest) (*MGetResponse, error)
+	Delete(path string) (string, error)
+	Put(path string, body string) (string, error)
+	Post(path string, body string) (string, error)
+}
+
+//DefaultElasticsearchClient is an admin client to query a local instance of Elasticsearch
+type DefaultElasticsearchClient struct {
+	serverURL string
+	client    *http.Client
+}
+
+//NewElasticsearchClient is the initializer to create an instance of ES client
+func NewElasticsearchClient(skipVerify bool, serverURL, adminCert, adminKey string, adminCA []string) (ElasticsearchClient, error) {
+	caCertPool := x509.NewCertPool()
+	for _, ca := range adminCA {
+		caCert, err := ioutil.ReadFile(ca)
+		if err != nil {
+			log.Fatal(err)
+		}
+		caCertPool.AppendCertsFromPEM(caCert)
+	}
+
+	cert, err := tls.LoadX509KeyPair(adminCert, adminKey)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	client := &http.Client{
+		Transport: &http.Transport{
+			TLSClientConfig: &tls.Config{
+				RootCAs:            caCertPool,
+				Certificates:       []tls.Certificate{cert},
+				InsecureSkipVerify: skipVerify,
```

Comment:
do we want this to be configurable? Shouldn't we always verify/trust the ES we're connecting to?

Reply:
This depends.  Technically if we ALWAYS deploy this in the same pod as Elasticsearch then we don't even really need to talk via SSL.  The plan is to configure ES for proxy auth and only trust localhost anyway. We don't really care. This may be in here now to satisfy my existing testing

Reply:
configurable from switches.  I would expect EO to set to false in the configs.  resolving for now as this facilitates testing locally